### PR TITLE
Feature - Export BusinessFlow as UseCase

### PR DESCRIPTION
**Pull request by IamRanjeetSingh**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs [hunk 1]
```diff
@@ -118,6 +132,7 @@ void ITreeViewItem.SetTools(ITreeView TV)
                 TreeViewUtils.AddSubMenuItem(ExportMenu, "Export to ALM", ExportToALM, null, eImageType.ALM);
                 TreeViewUtils.AddSubMenuItem(ExportMenu, "Map to ALM", MapToALM, null, eImageType.MapALM);
                 TreeViewUtils.AddSubMenuItem(ExportMenu, "Export to CSV", ExportToCSV, null, eImageType.CSV);
+                TreeViewUtils.AddSubMenuItem(ExportMenu, "Export to BPMN file", ExportBPMNMenuItem_Click, icon: eImageType.ShareExternal);
```
Comment (Maheshkale447) [inline Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs:135]: As we are creating ZIP file, please change name to 'Export as Otoma Use Case Zip file'

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs [hunk 2]
```diff
@@ -225,5 +240,93 @@ private void ExportToCSV(object sender, System.Windows.RoutedEventArgs e)
...
+
+                wasAllAddedToSharedRepository = activityGroups.All(ag => ag.IsSharedRepositoryInstance);
+                return wasAllAddedToSharedRepository;
+            }
+            catch(Exception ex)
+            {
+                Reporter.ToUser(eUserMsgKey.FailedToAddItemsToSharedRepository, "Unexpected error, check logs for more details");
+                Reporter.ToLog(eLogLevel.ERROR, "Error occurred while adding missing activities and activity groups to shared repository.", ex);
+                wasAllAddedToSharedRepository = false;
+                return wasAllAddedToSharedRepository;
```
Comment (Maheshkale447) [inline Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs:297]: Add return statement to Finally as we same flag is being returned from both try and catch block

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs [hunk 3]
```diff
@@ -225,5 +240,93 @@ private void ExportToCSV(object sender, System.Windows.RoutedEventArgs e)
...
+            }
+        }
+
+        private void ExportBusinessFlowBPMN()
+        {
+            try
+            {
+                Reporter.ToStatus(eStatusMsgKey.ExportingToBPMNFile);
```
Comment (Maheshkale447) [inline Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs:305]: Change status to 'Exporting as Otoma Use Case Zip file'

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs (hunk 2 above)
Comment (IamRanjeetSingh) [reply Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs:297]: As discussed, we won't do this modification

**Review by Maheshkale447 — APPROVED** [review]

**Review by Maheshkale447 — DISMISSED** [review]

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**